move/its/sources/service.move

```diff
@@ -95,19 +109,43 @@ module its::service {
             .write_u256((decimals as u256))
             .write_bytes(vector::empty());
 
-        send_payload(self, destination_chain, writer.into_bytes());
+        prepare_message(self, destination_chain, writer.into_bytes())
     }
 
-    public fun interchain_transfer<T>(
-        self: &mut ITS,
+    public fun prepare_interchain_transfer<T>(
         token_id: TokenId,
         coin: Coin<T>,
         destination_chain: String,
         destination_address: vector<u8>,
         metadata: vector<u8>,
         source_channel: &Channel,
+    ): InterchainTransferTicket<T> {
+        InterchainTransferTicket<T> {
+            token_id,
+            coin,
+            source_address: source_channel.to_address(),
+            destination_chain,
+            destination_address,
+            metadata,
+            version: VERSION,
+        }
+    }
+
+    public fun submit_interchain_transfer<T>(
```

Comment:
should be an entry fun

Reply:
sadly `entry` functions cannot receive hot potatoes